layer/logged_data/system_metrics.py

```diff
@@ -298,12 +298,12 @@ def __enter__(self) -> None:
             raise RuntimeError(
                 "System stats logging only allowed with a logged data destination"
             )
-        train = active_context.train()
-        train_id = train.get_id() if train is not None else None
+        model_train = active_context.model_train()
+        model_train_id = model_train.id if model_train is not None else None
         dataset_build = active_context.dataset_build()
         dataset_build_id = dataset_build.id if dataset_build is not None else None
         self._log_data_runner = LogDataRunner(
-            train_id=train_id,
+            train_id=model_train_id,
             dataset_build_id=dataset_build_id,
```

Comment:
this part we can generalize further in the future i feel

Reply:
absolutely. we may also just switch over to "artifact_id" and stop using dataset build and model train